### PR TITLE
feat(auth-manager): implemented connection resource including tests

### PR DESCRIPTION
**Pull request by CptSchnitz**

### CODE REVIEW

File: packages/auth-manager/src/connection/DAL/connectionRepository.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        .select('version')
+        .where('name = :name AND environment = :environment')
+        .andWhere((qb) => {
+          const subQuery = qb
+            .subQuery()
+            .select('MAX(version)')
+            .from(Connection, 'connection')
+            .where('name = :name AND environment = :environment')
+            .getQuery();
+
```
Comment (syncush) [inline packages/auth-manager/src/connection/DAL/connectionRepository.ts:25]: maybe external function?

File: packages/auth-manager/src/domain/DAL/domainRepository.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+export const domainRepositoryFactory: FactoryFunction<DomainRepository> = (container) => {
+  const dataSource = container.resolve(DataSource);
+
+  return dataSource.getRepository(Domain).extend({
+    async checkInputForNonExistingDomains(domainNames: string[]): Promise<string[]> {
+      // i wrote raw sql because typeorm doesn't think that using a function in FROM is a real thing and treats it like a table name
+      const res = (await this.manager.query(
+        `
+        SELECT i.name FROM unnest($1::text[]) i(name) LEFT JOIN auth_manager.domain d ON i.name = d.name WHERE d.name is NULL`,
```
Comment (syncush) [inline packages/auth-manager/src/domain/DAL/domainRepository.ts:17]: is it postgres only sql?

File: packages/auth-manager/tests/integration/domain/domain.spec.ts [hunk 3]
```diff
@@ -39,7 +39,7 @@ describe('domain', function () {
 
         expect(res).toHaveProperty('status', httpStatusCodes.OK);
         expect(res).toSatisfyApiSpec();
-        expect(res.body).toStrictEqual([{ name: 'avi' }, { name: 'iva' }]);
+        expect(res.body).toEqual(expect.arrayContaining([{ name: 'avi' }, { name: 'iva' }]));
```
Comment (syncush) [inline packages/auth-manager/tests/integration/domain/domain.spec.ts:42]: iva is avi's wife

**Review by syncush — CHANGES_REQUESTED** [review]

File: packages/auth-manager/src/domain/DAL/domainRepository.ts (hunk 2 above)
Comment (CptSchnitz) [reply packages/auth-manager/src/domain/DAL/domainRepository.ts:17]: yes

File: packages/auth-manager/src/connection/DAL/connectionRepository.ts (hunk 1 above)
Comment (CptSchnitz) [reply packages/auth-manager/src/connection/DAL/connectionRepository.ts]: extracted

File: packages/auth-manager/src/domain/DAL/domainRepository.ts (hunk 2 above)
Comment (syncush) [reply packages/auth-manager/src/domain/DAL/domainRepository.ts:17]: I would add that to the comment

**Review by syncush — APPROVED** [review]

**Comment by syncush** [comment]
I approved with a small comment for the sql

**Pull request merged**